test/evergreen/compatibility_test_for_releases.sh

```diff
@@ -327,6 +339,52 @@ top="test-compatibility-run"
 rm -rf "$top" && mkdir "$top"
 cd "$top"
 
+if [ "$upgrade_to_latest" = true ]; then
+    # Prepare test data.
+    echo "Preparing test data..."
+    git clone --quiet --depth 1 --filter=blob:none --no-checkout https://github.com/wiredtiger/mongo-tests.git
+    cd mongo-tests; git checkout --quiet master -- WT-8395; cd WT-8395; test_data=$(pwd)
+
+    for FILE in $test_data/*; do tar -zxf $FILE; done
+    rm *.tar.gz; cd ../..
+
+    test_root=$(pwd)
+    for b in ${upgrade_to_latest_upgrade_downgrade_release_branches[@]}; do
+        (build_branch $b)
+        cd $b/test/checkpoint
+
+        for FILE in $test_data/*; do
+            # Run actual test.
+            echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
+            echo " Upgrading $FILE database to $b..."
+            echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
+
+            # Disable exit on non 0. We expect ./t to fail in certain scenarios.
+            set +e
+
+            ./t -t r -D -v -h $FILE
+            test_res=$?
+
+            # Enable exit on non 0
+            set -e
+
+            # Validate test result.
+            if [[ "$FILE" =~ "4.4."[0-6]"_unclean"$ ]]; then
+                if [[ "$test_res" == 0 ]]; then
+                    echo "Error: Databases generated with unclean shutdown from versions 4.4.[0-6] must fail!"
+                    exit 1
+                fi
+            elif [[ "$test_res" != 0 ]]; then
+                echo "Error: Upgrade failed! Test result is $test_res."
+                exit 1
+            fi
+
+        done
+        cd $test_root
+    done
+
+    exit 0
```

Comment:
I think we this exit should be removed. If we `exit 0` here, any other options passed to the script will be ignored i.e. `-n`,`-o`. I believe the script is supposed be able to run all the test options in a single execution.

Reply:
Done!